crates/core/src/grpc.rs

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) 2025 Restate Software, Inc., Restate GmbH.
+// All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use futures::future::BoxFuture;
+use tonic::{body::BoxBody, server::NamedService};
+
+const VERSION_HEADER: &str = "X-Api-Version";
```

Comment:
Could you please add a doc comment defining the meaning of these headers, and how clients should behave when they encounter them?